### PR TITLE
Enable building kubectl hermetically as part of toolchain setup

### PR DESCRIPTION
**Pull request by nlopezgi**
Ready for review now.
This PR enables building kubectl. This is now the default behavior which can be overriden by using a custom kubectl_configure(name="k8s_config" ...) target. 
This PR resolves: https://github.com/bazelbuild/rules_k8s/issues/227

### CODE REVIEW

File: k8s/k8s.bzl [hunk 1]
```diff
@@ -44,4 +44,4 @@ py_library(
   )
 
   # WORKSPACE target to configure the kubectl tool
-  kubectl_configure(name = "local_k8s_config")
+  kubectl_build(name = "local_k8s_config")
```
Comment (smukherj1) [inline k8s/k8s.bzl]: Change 'local_k8s_config' to 'k8s_config' or 'k8s_build'. Also skip kubectl_build if the repository key exists in native.existing_rules().keys(). See https://github.com/bazelbuild/rules_docker/blob/9527234ef0b5a57bce93be524cb56d7ab1a85ea3/container/container.bzl#L227 for how we did this in rules_docker

Comment (nlopezgi) [reply k8s/k8s.bzl]: done, thanks for the reminders

File: k8s/object.bzl [hunk 2]
```diff
@@ -28,179 +28,186 @@ load(
...
+    if "{" in ctx.attr.namespace:
+        namespace_file = ctx.new_file(ctx.label.name + ".namespace-name")
+        _resolve(ctx, ctx.attr.namespace, namespace_file)
+        namespace_arg = "$(cat %s)" % _runfiles(ctx, namespace_file)
+        files += [namespace_file]
+
+    if namespace_arg:
+        namespace_arg = "--namespace=\"" + namespace_arg + "\""
+
+    kubectl_tool_info = ctx.toolchains["@io_bazel_rules_k8s//toolchains/kubectl:toolchain_type"].kubectlinfo
```
Comment (nlopezgi) [inline k8s/object.bzl]: Note this is the only thing that changed in this file, but buildifier decided to change lots of other things

Comment (chrislovecnm) [reply k8s/object.bzl]: Holy crap Batman

Comment (chrislovecnm) [reply k8s/object.bzl]: Why don’t you run buildifier as a separate PR?

Comment (nlopezgi) [reply k8s/object.bzl]: ugh, I think we have a check for some files to be properly formatted (or thought) and ran buildifier just to be on the safe side. Created https://github.com/bazelbuild/rules_k8s/pull/230 to just do buildifier; will rebase this once that is in. No need to review this PR until I have that PR in and the e2e tests passing. Thanks!

**Comment by nlopezgi** [comment]
@alex1545 fyi

File: toolchains/kubectl/kubectl_configure.bzl [hunk 3]
```diff
@@ -16,16 +16,79 @@ Defines a repository rule for configuring the kubectl tool.
...
+        k8s_commit = kwargs["k8s_commit"] if "k8s_commit" in kwargs else "v1.13.0-beta.1"
+        k8s_sha256 = kwargs["k8s_sha256"] if "k8s_sha256" in kwargs else "dfb39ce36284c1ce228954ca12bf016c09be61e40a875e8af4fff84e116bd3a7"
+        k8s_prefix = kwargs["k8s_prefix"] if "k8s_prefix" in kwargs else "kubernetes-1.13.0-beta.1"
+        http_archive(
+            name = "io_kubernetes",
+            sha256 = k8s_sha256,
+            strip_prefix = k8s_prefix,
+            urls = [("https://github.com/kubernetes/kubernetes/archive/%s.tar.gz" % k8s_commit)],
+        )
+        http_archive(
```
Comment (chrislovecnm) [inline toolchains/kubectl/kubectl_configure.bzl]: Do we need this?

File: toolchains/kubectl/kubectl_configure.bzl [hunk 4]
```diff
@@ -16,16 +16,79 @@ Defines a repository rule for configuring the kubectl tool.
...
+    properly configured.
+
+    Args:
+        **kwargs:
+      Required Args
+        name: A unique name for this rule.
+      Default Args:
+        build_srcs: Optional. Set to true to build kubectl from sources. Default: False
+        k8s_commit: Otional. Commit / release tag at which to build kubectl
+          from. Default "v1.13.0-beta.1"
```
Comment (chrislovecnm) [inline toolchains/kubectl/kubectl_configure.bzl]: we need an issue to update this when 1.13 is out.

Also we need docs that only = or gt 1.13 is supported

File: toolchains/kubectl/kubectl_configure.bzl (hunk 3 above)
Comment (nlopezgi) [reply toolchains/kubectl/kubectl_configure.bzl]: yes, this is what Jeff Grafton was referring to as problematic with the kubernetes repo and which would be solved by the abandoned cl.

File: toolchains/kubectl/kubectl_configure.bzl (hunk 4 above)
Comment (nlopezgi) [reply toolchains/kubectl/kubectl_configure.bzl]: will do docs in upcoming PR to avoid re-triggering CI.

**Pull request merged**